store/errors.go

```diff
@@ -61,11 +63,19 @@ var (
 
 	// ErrNoUpdateAvailable is returned when an update is attempetd for a snap that has no update available.
 	ErrNoUpdateAvailable = errors.New("snap has no updates available")
-
-	// ErrRevisionNotAvailable is returned when an install is attempted for a snap but the/a revision is not available (given install constraints)
-	ErrRevisionNotAvailable = errors.New("no snap revision given constraints")
 )
 
+// RevisionNotAvailableError is returned when an install is attempted for a snap but the/a revision is not available (given install constraints).
+type RevisionNotAvailableError struct {
+	Action   string
+	Channel  string
+	Releases []snap.Channel
+}
+
+func (e *RevisionNotAvailableError) Error() string {
+	return "no snap revision given constraints"
```

Comment:
Was already there before, but this reads slightly strange somehow.  Perhaps because "no snap revision given" is a sentence by itself and with a different meaning.

Perhaps:

```
"no snap revision available as specified"
```

It's also a bit curious that we have the metadata for a rich error to be produced here, but we opt to just send a general message.

Reply:
done